src/crawlee/sessions/_session.py

```diff
@@ -193,17 +193,22 @@ def is_blocked_status_code(
         self,
         *,
         status_code: int,
-        additional_blocked_status_codes: list[int] | None = None,
+        additional_blocked_status_codes: set[int] | None = None,
+        ignore_http_error_status_codes: set[int] | None = None,
     ) -> bool:
         """Evaluate whether a session should be retired based on the received HTTP status code.
 
         Args:
             status_code: The HTTP status code received from a server response.
             additional_blocked_status_codes: Optional additional status codes that should trigger session retirement.
+            ignore_http_error_status_codes: Optional status codes to allow suppression of
+            codes from `blocked_status_codes`.
 
         Returns:
             True if the session should be retired, False otherwise.
         """
-        blocked_status_codes = self._blocked_status_codes + (additional_blocked_status_codes or [])
+        blocked_status_codes = self._blocked_status_codes | (additional_blocked_status_codes or set())
+
+        blocked_status_codes = blocked_status_codes - (ignore_http_error_status_codes or set())
```

Comment:
Not sure if I'm reading this right, but it seems to me that `ignore_http_error_status_codes` takes precedence over `additional_blocked_status_codes` here, but in `BaseHttpClient`, the `additional_http_error_status_codes` option takes precedence over `ignore_http_error_status_codes`.

This is an edge case for sure, but still, I'd like to keep things consistent.

Reply:
You're right.

Updated the code to make `additional_http_error_status_codes` take precedence in both cases, which matches the behavior in the JS version